synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -1498,3 +1506,96 @@ def _populate_txn(txn: LoggingTransaction) -> bool:
             )
 
         return batch_size
+
+    async def _populate_membership_event_stream_ordering(
+        self, progress: JsonDict, batch_size: int
+    ) -> int:
+        def _populate_membership_event_stream_ordering(
+            txn: LoggingTransaction,
+        ) -> bool:
+            if "max_stream_ordering" in progress:
+                max_stream_ordering = progress["max_stream_ordering"]
+            else:
+                txn.execute("SELECT max(stream_ordering) FROM events")
+                res = txn.fetchone()
+                if res is None or res[0] is None:
+                    return True
+                else:
+                    max_stream_ordering = res[0]
+
+            start = progress.get("stream_ordering", 0)
```

Comment:
note that it's possible for events to have a negative `stream_ordering`, and this update looks like it will miss them out. One way to solve this is what we did for `events_populate_state_key_rejections`: the `INSERT INTO background_updates` statement inserts the start point.